src/instruments/src/Common/simVars.tsx

```diff
@@ -0,0 +1,455 @@
+type RegisterSimVar = (name: string, unit: UnitName, maxStaleness: number, global: boolean) => void;
+type UnregisterSimVar = (name: string, unit: UnitName, maxStaleness: number, global: boolean) => void;
+
+const errorCallback = () => {
+    throw Error('useSimVar was called in a React tree with no SimVarProvider');
+};
+const context = React.createContext<{
+    retrieve: RetrieveSimVar,
+    update: UpdateSimVar,
+    register: RegisterSimVar,
+    unregister: UnregisterSimVar,
+}>({
+    retrieve: errorCallback,
+    update: errorCallback,
+    register: errorCallback,
+    unregister: errorCallback,
+});
+const {Provider: InternalProvider}  = context;
+
+type UnitName = string | any; // once typings is next to tsconfig.json, use those units
+type SimVarValue = number | any;
+type SimVarCache = Record<string, {
+    value: SimVarValue,
+    lastUpdatedAgo: number,
+}>;
+
+declare const SimVar; // this can also be replaced once /typings are available
+
+/**
+ * This component provides the basic functionality for the useSimVar hooks.
+ * By keeping the last known SimVar values inside this provider, we're
+ * effectively caching each SimVar, so that there is no additional overhead when
+ * using multiple useSimVar hooks for the same SimVar.
+ * For improved performance, this component will only trigger renders when the
+ * "update" custom event is emitted through an instrument.
+ */
+const SimVarProvider: React.FC = ({ children }) => {
+    const listeners = React.useRef<Record<string, number[]>>({});
+    const [cache, setCache] = useState<SimVarCache>({});
+
+    useUpdate((deltaTime: number) => {
+        const stateUpdates: Record<string, {
+            value?: SimVarValue,
+            lastUpdatedAgo: number,
+        }> = {};
+
+        for (const [key, intervals] of Object.entries(listeners.current)) {
+            // First, let's check if there are any listeners at all
+            if (!intervals.length) {
+                continue;
+            }
+
+            // The refresh time is given by the *smallest* maximum update
+            // interval.
+            const threshold = Math.min(...intervals);
+            const lastUpdatedAgo = (cache[key] ? cache[key].lastUpdatedAgo || 0 : 0) + deltaTime;
+
+            if (lastUpdatedAgo >= threshold) {
+                // At this point, as we haven't updated this SimVar recently, we
+                // need to fetch the latest value from the simulator and store
+                // it.
+                const [name, rawUnit] = key.split('/');
+                const unit = normalizeUnitName(rawUnit as UnitName);
+                stateUpdates[key] = {
+                    value: name.startsWith('_GLOBAL_')
+                        ? SimVar.GetGlobalVarValue(name.substr(8), unit)
+                        : SimVar.GetSimVarValue(name, unit),
+                    lastUpdatedAgo: lastUpdatedAgo % threshold,
+                };
```

Comment:
**suggestion**: Again if performance is the absolute goal: consider storing (1) the normalised unit, (2) whether it's global or not, and (3) the name in a separate object alongside the key, such that there is no need to parse it every time.